### PR TITLE
[tree] Apply suggestions from clang-tidy

### PR DESCRIPTION
**Pull request by guitargeek**
Applies the clang-tidy suggestions from the following checks, which are the same as in the CMSSW `.clang-tidy`, only that the checks that replace copies with `const` references are removed:

```
  ,boost-use-to-string,
  ,misc-definitions-in-headers,
  ,misc-string-compare,
  ,misc-uniqueptr-reset-release,
  ,modernize-deprecated-headers,
  ,modernize-make-shared,
  ,modernize-make-unique,
  ,modernize-use-bool-literals,
  ,modernize-use-equals-delete,
  ,modernize-use-nullptr,
  ,modernize-use-override,
  ,performance-inefficient-algorithm,
  ,performance-inefficient-vector-operation,
  ,performance-faster-string-find,
  ,performance-move-const-arg,
  ,readability-container-size-empty,
  ,readability-redundant-string-cstr,
  ,readability-static-definition-in-anonymous-namespace,
  ,readability-uniqueptr-delete-release
```

### CODE REVIEW

**Review by jblomer — APPROVED** [review]
Many thanks!  Looks good to me for the RNTuple part and the SQlite RDF data source.

I leave it up to the other code owners to approve the other parts.

**Comment by guitargeek** [comment]
@eguiraud, @pcanal, is it okay for the other parts?

File: tree/dataframe/src/RDFHelpers.cxx
Comment (eguiraud) [inline tree/dataframe/src/RDFHelpers.cxx:358]:
```suggestion
   ROOT::RDF::SampleCallback_t GetSampleCallback() final
```

File: tree/dataframe/src/RDFInterfaceUtils.cxx
Comment (eguiraud) [inline tree/dataframe/src/RDFInterfaceUtils.cxx:95]: here and below, the extra `static` was something that @pcanal or @Axel-Naumann suggested at some point as the way we write this in ROOT -- I don't remember exactly why, I think it had to do with MSVC++?

I'm ok with removing it but let's ping them to be sure.

File: tree/dataframe/src/lexertk.hpp
Comment (eguiraud) [inline tree/dataframe/src/lexertk.hpp:777]: I would leave `lexertk.hpp` alone, we are just vendoring this third-party project, we are not actually maintaining the code. applying changes like these might make it harder to update it or patch it.

on the other hand, if we are never going to update it or patch it, we might as well treat it as our own code. up to @dpiparo and @vepadulano I guess.

File: tree/dataframe/test/dataframe_simple.cxx
Comment (eguiraud) [inline tree/dataframe/test/dataframe_simple.cxx:640]: way to go clang-tidy! :)

**Review by eguiraud — APPROVED** [review]
reviewed the RDF part, see comments

File: tree/dataframe/src/RDFInterfaceUtils.cxx
Comment (Axel-Naumann) [reply tree/dataframe/src/RDFInterfaceUtils.cxx:95]: For Windows: on the contrary, static members are problematic across DLLs.

Could have been because of cling? I really don't know, `git blame` might be helpful here!

If the CI is happy then this looks like a good change to me.

File: tree/dataframe/src/lexertk.hpp
Comment (guitargeek) [reply tree/dataframe/src/lexertk.hpp:777]: You are right! It has been also our policy in similar situations to not change the file, so I will revert the changes to this file in this PR

File: tree/dataframe/src/RDFInterfaceUtils.cxx
Comment (guitargeek) [reply tree/dataframe/src/RDFInterfaceUtils.cxx:95]: Thanks for the discussion! The CI is happy, so let's try to remove the `static` in these situations, like `clang-tidy` suggests.

**Comment by guitargeek** [comment]
Thanks for your review, @eguiraud! I addressed your requests about not changing the lexertk file and replacing the `override` with `final` in one case.

**Pull request merged**
